### ISSUE TITLE
Incomplete class names for classes used in Model when generating mixin

### ISSUE THREAD

**Issue by ajay-mm** [barryvdh/laravel-ide-helper#1173]
### Versions:
- ide-helper Version: 2.9.0
- Laravel Version: 8.27.0
- PHP Version: 7.3.27

### Description:

When generating phpdoc for models as a mixin (`php artisan ide-helper:models -M`), the full class names are not generated if the class file contains a 'use statement' for that class. This results in autocomplete failure and linter warnings.

### Steps To Reproduce:

1. Add a use statement for Builder class in any existing model
`use Illuminate\Database\Eloquent\Builder;`

2. Run `php artisan ide-helper:models -M` to generate `_ide_helper_models.php`

3. Check the generated file for Model mixin. Only simple class names would be included instead of full class name.

```php
 * @method static Builder|Model newModelQuery()
 * @method static Builder|Model newQuery()
 * @method static Builder|Model query()
```

### Expected output

The mixin should have full paths to the class names
```php
 * @method static \Illuminate\Database\Eloquent\Builder|Model newModelQuery()
 * @method static \Illuminate\Database\Eloquent\Builder|Model newQuery()
 * @method static \Illuminate\Database\Eloquent\Builder|Model query()
```

### Notes

The model phpdoc generation uses full classnames if `--nowrite (-N)` option is used to generate the `_ide_helper_models.php`.

**Comment by rovansteen** [comment]
I ran into the same issue and while I would consider this a bug and the FQN should always be used when using the `--write-mixin/-M` flag there is a feasible workaround by setting `'force_fqn' => true` in the config.

**Comment by ajay-mm** [comment]
Thanks @rovansteen for the quick workaround.

I too think FQN should be used with write-mixin. I went through the code and found the source of the bug. When the write-mixin option was added (#764), check for writing to external file was updated in the `handle()` method but missed in the `getClassNameInDestinationFile()`.

The fix is simple.

https://github.com/barryvdh/laravel-ide-helper/blob/b21324b20e24a3c36bedc184b4cf273bfa0a2d32/src/Console/ModelsCommand.php#L1155 should be changed to 

```php
        $writingToExternalFile = !$this->write || $this->write_mixin;
```
So that it matches the check for writing to external file in handle method.

https://github.com/barryvdh/laravel-ide-helper/blob/b21324b20e24a3c36bedc184b4cf273bfa0a2d32/src/Console/ModelsCommand.php#L164

I have verified that this change fixes the bug on my machine.
